src/kps/business/RouteCalculator.java

```diff
@@ -10,12 +10,13 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class RouteCalculator {
     private List<Route> routes;
 
     public RouteCalculator(Simulation simulation) {
-        this.routes = simulation.getRoutes();
+        this.routes = simulation.getUniqueRoutes().stream().filter(r -> !r.isDiscontinued()).collect(Collectors.toList());
```

Comment:
There are 3 routes from Wellington and Sydney, but the getUniqueRoutes only gets the first one which is discontinued. Then the filter removes the route from the possible routes leaving no routes between Wellington and Sydney. This is why the tests are failing to find a route from Auckland to Suva.